src/compiler/parser.ts

```diff
@@ -1105,6 +1121,9 @@ module ts {
         // attached to the EOF token.
         let parseErrorBeforeNextFinishedNode: boolean = false;
 
+        let inJSXTag = false;

```

Comment:
Yes it is only for identifier scanning in xml tags. I will try to move the validation of this identifier in the parser.